devdoc/waiter.md

```diff
@@ -0,0 +1,231 @@
+# `waiter` requirements
+
+## Overview
+
+`waiter` is a module that allows the user to move data asynchronously from producer to consumer without having to block the thread.
```

Comment:
>move data asynchronously from producer to consumer [](http://example.com/codeflow?start=45&length=50)

Poetry:

nothing here "waits". So the module being called "waiter" does not convey what "it does". Consider other poetry. Here's some other possible names: "buffer", "broker", "channel"

Reply:
I like channel. Reminds me of Golang channels